Feature: Address space update in peered vNets

harics24 opened the pull request:
Adding new fields and operation to support the address space update in peered vNets

These changes were reviewed earlier and had to be reverted as the backend changes were not yet ready back then.
Ref PR: #11090

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [x] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [X] SDK of .NET (need service team to ensure code readiness)
      - [X] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [X] PowerShell
      - [X] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [X] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [X] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [X] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [X] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [X] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

Commit:
Adding new fields and operation to support the address space update in peered vNets

Adding new fields and operation to support the address space update in peered vNets

harics24 commented:
Can someone review this and approve the changes?
@MikhailTryakhov , is it ok to ignore the 'NetworkValidation'?

MikhailTryakhov commented:
@kairu-ms  can you please help to find out why is autorest version is incorrect? But actually I think we need to sync up with .net sdk team to ask when would it be supported on the latest autorest...
@harics24 this is optional job, so yeah you can skip it.

kairu-ms commented on specification/network/resource-manager/Microsoft.Network/stable/2021-02-01/virtualNetwork.json:
```diff
@@ -1038,6 +1038,13 @@
             "description": "Parameters supplied to the create or update virtual network peering operation."
           },
+          {
+            "name": "syncRemoteAddressSpace",
```
I found this parameter is not used in examples. Can you add usage of it?

harics24 replied:
Added it now. It's a query parameter. Thanks for catching this.

harics24 replied:
The way we handle the query param should work in either way. We are not using quotes in the query params too. I have pinged you separately with more details. Please let me know if I need to provide the values in example to be a string, like "true".

kairu-ms replied:
> The way we handle the query param should work in either way. We are not using quotes in the query params too. I have pinged you separately with more details. Please let me know if I need to provide the values in example to be a string, like "true".

Swagger ModelValidation is failing because of that. I think you should use string here.

kairu-ms replied:
BTW, this property should be enum, as its value could only be "true" or "false". Like following

https://github.com/Azure/azure-rest-api-specs/blob/e4c056151df3c59bbb2221e23f623fdb625313c5/specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2017-03-01/apimapis.json#L3489-L3502

kairu-ms commented:
@

> @kairu-ms can you please help to find out why is autorest version is incorrect? But actually I think we need to sync up with .net sdk team to ask when would it be supported on the latest autorest...

Hi @lirenhe, can you help with this problem. Thanks.

Commit:
Adding the new query param in the example

As per the review comment, adding the new query param in the example request response of swagger.

Commit:
Adding the new query param in examples

Adding the new query param in examples

kairu-ms commented on specification/network/resource-manager/Microsoft.Network/stable/2021-02-01/virtualNetwork.json:
```diff
@@ -1038,6 +1038,13 @@
+            "name": "syncRemoteAddressSpace",
+            "in": "query",
+            "required": false,
+            "type": "string",
```
The type is string, however it's bool value in example. Why?

Commit:
Restricting the sync param

Restricting the sync param to hold only true as value. We never need to send false.

harics24 commented:
> Hi @harics24, Your PR has some issues. Please fix the CI sequentially by following the order of `Avocado, semantic validation, model validation, breaking change, lintDiff`.
> 
> Task	How to fix	Priority	Support (Microsoft alias)
> Avocado	[Fix-Avocado](https://github.com/Azure/avocado)	High	ruowan
> Semantic validation	[Fix-SemanticValidation-Error](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/ci-fix.md#semantic-validation)	High	raychen, jianyxi
> Model validation	[Fix-ModelValidation-Error](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/ci-fix.md#model-validation)	High	raychen,jianyxi
> LintDiff	[Fix-LintDiff](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/ci-fix.md#linter-validation)	high	jianyxi, ruoxuan
> If you need further help, please feedback via [swagger feedback](https://aka.ms/swaggerfeedback)."

I had run, except LintDiff, everything is suceeded.

Even in LintDiff, it did not fail for my changes.
